### PR TITLE
Build: Update test code for compatibility with QUnit 2.x

### PR DESCRIPTION
**Pull request by mgol**
### Summary ###

Also, run `grunt npmcopy` to sync the "external" directory with dependencies
from package.json. For example, the Sinon library version didn't match.

This is gh-4234 without the QUnit upgrade & with some additional fixes to make it all work. I verified it works on my local TestSwarm setup.

We can do the actual QUnit upgrade later with all the needed final fixes.

Ref gh-4234

### Checklist ###

* [x] All authors have signed the CLA at https://cla.js.foundation/jquery/jquery
* ~~New tests have been added to show the fix or feature works~~
* [x] Grunt build and unit tests pass locally with these changes
* ~~If needed, a docs issue/PR was created at https://github.com/jquery/api.jquery.com~~

### CODE REVIEW

File: test/data/inner_module.js [hunk 1]
```diff
@@ -1 +1 @@
-window.ok( true, "evaluated: inner module with src" );
```
Comment (dmethvin) [inline test/data/inner_module.js]: Whoa, that kind of makes it clear we were depending on an `ok()` function in global scope. Scary.

File: external/requirejs/require.js [hunk 2]
```diff
@@ -1664,13 +1671,12 @@ var requirejs, require, define;
 
                     //Join the path parts together, then figure out if baseUrl is needed.
                     url = syms.join('/');
-                    url += (ext || (/^data\:|\?/.test(url) || skipExt ? '' : '.js'));
+                    url += (ext || (/^data\:|^blob\:|\?/.test(url) || skipExt ? '' : '.js'));
```
Comment (dmethvin) [inline external/requirejs/require.js]: Does this change belong with the rest? It doesn't seem to be related to QUnit.

Comment (mgol) [reply external/requirejs/require.js]: @dmethvin This + Sinon changes is just running `npmcopy` on what we had. We now have `external` matching what's in `node_modules`. This required some minor changes in test code.

Comment (dmethvin) [reply external/requirejs/require.js]: Oh duh, i didn't notice this file was in `/external`  with the others. Carry on!

**Comment by mgol** [comment]
GitHub messed up PR authors (changing it to me) so I immediately ammended the commit & force-pushed to c3498187421d0a50ee0369163428481137a04b99.
